database/src/pool/postgres.rs

```diff
@@ -1413,6 +1413,114 @@ where
             .await
             .unwrap();
     }
+
+    async fn get_benchmark_requests_by_status(
+        &self,
+        statuses: &[BenchmarkRequestStatus],
+    ) -> anyhow::Result<Vec<BenchmarkRequest>> {
+        // There is a small period of time where a try commit's parent_sha
+        // could be NULL, this query will filter that out.
+        let query = "
+                SELECT
+                    tag,
+                    parent_sha,
+                    pr,
+                    commit_type,
+                    status,
+                    created_at,
+                    completed_at,
+                    backends,
+                    profiles
+                FROM benchmark_request
+                WHERE status = ANY($1)
+                AND (commit_type <> 'try' OR parent_sha IS NOT NULL) "
+            .to_string();
+
+        let rows = self
+            .conn()
+            .query(&query, &[&statuses])
+            .await
+            .context("Failed to get benchmark requests")?;
+
+        let benchmark_requests = rows
+            .iter()
+            .map(|row| {
+                let tag = row.get::<_, &str>(0);
+                let parent_sha = row.get::<_, Option<String>>(1);
```

Comment:
`&str` can be used also for `parent_sha`, `commit_type`, `backends` and `profiles`.

Reply:
b3a37f305c796d711357a1cf0eb8781c72ba0dd5